.github/workflows/pr.yml

```diff
@@ -30,16 +41,85 @@ jobs:
             ${{ runner.os }}-yarn-
 
       - name: Install dependencies
-        run: yarn install --frozen-lockfile
+        run: yarn install --frozen-lockfile --ignore-engines
 
       - name: Build packages
         run: yarn build
 
-      # - name: Build CRA templates
-      #   run: yarn build:templates ../cra-templates
-
       - name: Lint types and code
         run: yarn lint
 
       - name: Run tests
         run: yarn test:ci
+
+  release:
+    name: Release Snapshot (if labelled)
+    runs-on: ubuntu-latest
+    if: contains( github.event.pull_request.labels.*.name, 'release:canary' )
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get Yarn cache path
```

Comment:
I wouldn't rush to introduce caching. It has its downsides and if the speed gain is only a few seconds, I would reconsider.